x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/IdentifierGenerator.java

```diff
@@ -65,6 +65,11 @@ public static String randomIndexPattern(Feature... features) {
             var cluster = randomIdentifier();
             pattern = maybeQuote(cluster + ":" + pattern);
         }
+
+        while (pattern.contains("|") && pattern.contains("\"") == false) {
+            pattern = maybeQuote(pattern);
```

Comment:
`maybeQuote` can randomly avoid quoting the pattern, but without qoutes the pattern would be invalid. I think we better have another method `randomQuotes` or so, that randomly applies either `"` or `"""` qutoes.

Reply:
Now the solution runs a random number of retries to end up certainly quoting anyway, which I think is odd and will surprise future readers. I'd have just added a method `randomQuote(pattern)` that applies either `"` or `"""` quoting, randomly, and would've used that inside of `maybeQuote` - by randomly either calling `randomQuote` or leaving this as-is. Then we'd use `randomQuote` here instead of an indeterminate number of calls to `maybeQuote`.